core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -513,60 +502,59 @@ private void registerBuiltinFunctionScoreParsers() {
     }
 
     private void registerBuiltinQueryParsers() {
-        registerQuery(MatchQueryBuilder.PROTOTYPE::readFrom, MatchQueryBuilder::fromXContent, MatchQueryBuilder.NAME,
-                "match_phrase", "matchPhrase", "match_phrase_prefix", "matchPhrasePrefix", "matchFuzzy", "match_fuzzy", "fuzzy_match");
-        registerQueryParser(MultiMatchQueryParser::new);
-        registerQueryParser(NestedQueryParser::new);
-        registerQueryParser(HasChildQueryParser::new);
-        registerQueryParser(HasParentQueryParser::new);
-        registerQueryParser(DisMaxQueryParser::new);
-        registerQueryParser(IdsQueryParser::new);
-        registerQueryParser(MatchAllQueryParser::new);
-        registerQueryParser(QueryStringQueryParser::new);
-        registerQueryParser(BoostingQueryParser::new);
+        registerQuery(MatchQueryBuilder.PROTOTYPE::readFrom, MatchQueryBuilder::fromXContent, MatchQueryBuilder.NAME, "match_phrase",
+                "match_phrase_prefix", "match_fuzzy", "fuzzy_match");
+        registerQuery(MultiMatchQueryBuilder.PROTOTYPE::readFrom, MultiMatchQueryBuilder::fromXContent, MultiMatchQueryBuilder.NAME);
```

Comment:
I don't think it makes much sense to keep then in the parser though. I can
make them a constant in the builder?
On Apr 1, 2016 7:04 AM, "Christoph Büscher" <notifications@github.com>
wrote:

> In core/src/main/java/org/elasticsearch/search/SearchModule.java
> <https://github.com/elastic/elasticsearch/pull/17458#discussion_r58191186>
> :
>
> > @@ -513,60 +502,59 @@ private void registerBuiltinFunctionScoreParsers() {
> >      }
> >
> >      private void registerBuiltinQueryParsers() {
> > -        registerQuery(MatchQueryBuilder.PROTOTYPE::readFrom, MatchQueryBuilder::fromXContent, MatchQueryBuilder.NAME,
> > -                "match_phrase", "matchPhrase", "match_phrase_prefix", "matchPhrasePrefix", "matchFuzzy", "match_fuzzy", "fuzzy_match");
> > -        registerQueryParser(MultiMatchQueryParser::new);
> > -        registerQueryParser(NestedQueryParser::new);
> > -        registerQueryParser(HasChildQueryParser::new);
> > -        registerQueryParser(HasParentQueryParser::new);
> > -        registerQueryParser(DisMaxQueryParser::new);
> > -        registerQueryParser(IdsQueryParser::new);
> > -        registerQueryParser(MatchAllQueryParser::new);
> > -        registerQueryParser(QueryStringQueryParser::new);
> > -        registerQueryParser(BoostingQueryParser::new);
> > +        registerQuery(MatchQueryBuilder.PROTOTYPE::readFrom, MatchQueryBuilder::fromXContent, MatchQueryBuilder.NAME, "match_phrase",
> > +                "match_phrase_prefix", "match_fuzzy", "fuzzy_match");
> > +        registerQuery(MultiMatchQueryBuilder.PROTOTYPE::readFrom, MultiMatchQueryBuilder::fromXContent, MultiMatchQueryBuilder.NAME);
>
> The previous builder also supported "multiMatch". I'm late to the party,
> but this is one problem I see in removing the alternative names() method
> from the builders, it moves the parser names (and alternatives) far away
> from the builders themselves. I guess this move is part of makign
> QueryParse a functional interface, but couldn't we leave the names()
> method and use it even though it will not be part of the QueryParser()
> interface? Or add it to AbstractQueryBuilder instead? Having all these
> string constants here in this class without a connection to their builder
> is a source of errors IMHO.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/17458/files/092a90dc36ed1712d3d9c97e1f3eb6deba0e348e#r58191186>
>